numpyro/contrib/hsgp/laplacian.py

```diff
@@ -5,34 +5,111 @@
 This module contains functions for computing eigenvalues and eigenfunctions of the laplace operator.
 """
 
+from __future__ import annotations
+
 from jaxlib.xla_extension import ArrayImpl
 
+import jax
 import jax.numpy as jnp
 
 
-# TODO: Adapt to dim >= 1.
-def sqrt_eigenvalues(ell: float, m: int) -> ArrayImpl:
+def eigenindices(m: list[int] | int, dim: int) -> ArrayImpl:
```

Comment:
Could we add a unit-test for this function? It would be great for documenting how is supposed to work.

Reply:
Added!